Fix part of #5134: Adding tests for core.controllers.custom_landing_pages.

rahgurung opened the pull request:
<!--
  - Thanks for submitting code to Oppia! Please fill out the following as part of
  - your pull request so we can review your code more easily.
  -->

## Explanation
<!--
  - Explain what your PR does. If this PR fixes an existing bug, please include
  - "Fixes #bugnum:" in the explanation so that GitHub can auto-close the issue
  - when this PR is merged.
  -->
Fixes part of #5134:  implements the tests for core.controllers.custom_landing_pages.
## Checklist
- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The linter/Karma pre-submit checks have passed.
  - These should run automatically, but if not, you can manually trigger them locally using `python scripts/pre_commit_linter.py` and `bash scripts/run_frontend_tests.sh`.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR follows the [style guide](https://github.com/oppia/oppia/wiki/Coding-style-guide).
- [x] The PR is assigned to an appropriate reviewer.
  - If you're a new contributor, please ask on [Gitter](https://gitter.im/oppia/oppia-chat) for someone to assign a reviewer.
  - If you're not sure who the appropriate reviewer is, please assign to the issue's "owner" -- see the "talk-to" label on the issue.

rahgurung committed:
Fix part of oppia#5134: adding tests for core.controllers.custom_landing_pages

DubeySandeep commented on core/controllers/custom_landing_pages_test.py:
```diff
@@ -0,0 +1,54 @@
+
+    def test_fraction_landing_page_without_viewer_type(self):
+        """Test for showing the landing page for fractions,
+        without any viewer type should redirect to Teacher type.
```
Teacher --> teacher's page.

DubeySandeep replied:
Also, can we test this when viewer type is available?

rahgurung replied:
Hey, actually I tried it but py linting was failing, I was doing something like 
`response = self.testapp.get(feconf.FRACTIONS_LANDING_PAGE_URL, {viewerType: 'Teacher'})`
Test passed but py linting failed, any suggestions on passing viewerType ?

rahgurung replied:
I referred to codebase and found a method to do so.

DubeySandeep commented on core/controllers/custom_landing_pages_test.py:
```diff
@@ -0,0 +1,54 @@
+
+
+class StewardsLandingPageTest(test_utils.GenericTestBase):
+    """Test for showing the landing page for stewards(parents, teachers,
```
Make `stewards(` --> `stewards (` (Add space)

rahgurung replied:
done

DubeySandeep reviewed (changes requested):
Hi @gurungrahul2, Thanks for the PR, I have left a few comments PTAL!

Also, in all the cases can we test some contents of the landing page? (as tested over [here](https://github.com/oppia/oppia/blob/ed1350e176a6cefcb469d4dd794b4fb4800181b3/core/controllers/pages_test.py#L27))

rahgurung committed:
Feedback: tested response content of all pages and added test for given viewertype

rahgurung committed:
fixed docstring

rahgurung committed:
fixed response must contain error

mustcontain was throwing error
so i removed strings with dot
and comma.

rahgurung committed:
fixed mustcontain error

used a class name as string
for testing as all the stewards
pages is provided by a single
html file.

DubeySandeep commented:
Hi @gurungrahul2, Can you please reply to the comments which are resolved? (As asked [here](https://github.com/oppia/oppia/wiki/Contributing-code-to-Oppia#address-review-comments-until-all-reviewers-give-lgtm-looks-good-to-me) in the 5th bullet.)

DubeySandeep commented:
Also, can you please update the PR tile as asked in the "checklist"?

gurungrahul2 commented:
@DubeySandeep  so what should be the PR title be ?

rahgurung commented:
Hey @DubeySandeep I made all required changes.

DubeySandeep commented on core/controllers/custom_landing_pages_test.py:
```diff
@@ -0,0 +1,69 @@
+        response = self.testapp.get(feconf.CUSTOM_NONPROFITS_LANDING_PAGE_URL)
+        self.assertEqual(response.status_int, 200)
+        self.assertEqual(response.content_type, 'text/html')
+        response.mustcontain('oppia-stewards-landing-section')
```
This LGTM, but can you please tell me what was the issue with `Contact us to arrange a pilot with your students.`?

rahgurung replied:
Tests were passing for me locally with that, but Travis gave me a red flag. I am not sure what happened but let me guess, can it be due to having a full stop (.) or question mark(?) in the old string? I am not sure.

rahgurung replied:
so what do you suggest , how to proceed with this ?

rahgurung replied:
@DubeySandeep

DubeySandeep replied:
@gurungrahul2, can you add a link to the file where  "Contact us to arrange a pilot with your students" is written? (I can't seem to find it!)

rahgurung replied:
I got it, a latest commit on 25 Nov changes the text thus my tests were failing.  13ab98e557

rahgurung replied:
@DubeySandeep I will adapt my changes and add a commit soon.

rahgurung replied:
done @DubeySandeep

DubeySandeep commented:
@gurungrahul2, Change the PR title to `Fix part of #bugno.: Adding tests for core.controllers.custom_landing_pages.`

rahgurung commented:
> @gurungrahul2, Change the PR title to `Fix part of #bugno.: Adding tests for core.controllers.custom_landing_pages.`

done already

rahgurung commented:
until you review, can you assign me this file so nobody else works on it and have conflicts? Thanks. @DubeySandeep

rahgurung committed:
adapted string to latest changes

old tests were failing because i was missing a commit 13ab98e
which was changing the text of
landing pages.

DubeySandeep reviewed (commented):
@gurungrahul2, Thanks for the changes! LGTM.

Can you please add a screenshot of the updated coverage report for landing pages controller? I'll approve the PR once it's 100%.

Thanks! :)

rahgurung commented:
![screenshot from 2018-12-10 15-05-49](https://user-images.githubusercontent.com/30543444/49723695-7ba80f80-fc8d-11e8-8821-05762e162526.png)

rahgurung commented:
@DubeySandeep here we go :)

DubeySandeep reviewed (approved):
LGTM! :)

DubeySandeep commented:
@gurungrahul2, congrats on your first PR to Oppia. :)

/cc @nithusha21
